osu.Game/Database/BeatmapDatabase.cs

```diff
@@ -139,9 +139,12 @@ public Beatmap GetBeatmap(BeatmapInfo beatmapInfo)
             return connection.GetAllWithChildren<T>(filter, recursive);
         }
         
-        public void GetChildren<T>(T item, bool recursive = true)
+        public T GetChildren<T>(T item, bool recursive = true)
         {
-            connection.GetChildren<T>(item, recursive);
+            if (item == null) return default(T);
+
+            connection.GetChildren(item, recursive);
+            return item;

```

Comment:
I feel like this change is misleading. It implies that you're getting a new object back and that this is a pure function, neither of which is the case.